modules/tinymce/src/core/main/ts/api/util/EventDispatcher.ts

```diff
@@ -112,7 +113,7 @@ class EventDispatcher<T> {
   private readonly settings: Record<string, any>;
   private readonly scope: {};
   private readonly toggleEvent: (name: string, toggle: boolean) => void;
-  private bindings = {};
+  private bindings = {} as { [K in string]?: { func: (event: EditorEvent<MappedEvent<T, K>>) => void; once?: true }[]; };
```

Comment:
This shouldn't be a cast. This is also quite complex, so can you break it out? eg:
```
type MappedBindings = { [K in string]?: Array<{ func: (event: EditorEvent<MappedEvent<T, K>>) => void; once?: true }> };
...
  private bindings: MappedBindings = {};
```
private Record<string,